edge-modules/load-gen/Program.cs

```diff
@@ -30,42 +32,45 @@ static async Task Main()
                     ModuleUtil.DefaultTransientRetryStrategy,
                     Logger);
 
-                using (var timers = new Timers())
-                {
-                    Guid batchId = Guid.NewGuid();
-                    Logger.LogInformation($"Batch Id={batchId}");
-
-                    // setup the message timer
-                    timers.Add(
-                        Settings.Current.MessageFrequency,
-                        Settings.Current.JitterFactor,
-                        () => GenerateMessageAsync(moduleClient, batchId));
-
-                    // setup the twin update timer
-                    timers.Add(
-                        Settings.Current.TwinUpdateFrequency,
-                        Settings.Current.JitterFactor,
-                        () => GenerateTwinUpdateAsync(moduleClient, batchId));
-
-                    timers.Start();
-                    (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
-                    Logger.LogInformation("Load gen running.");
-
-                    await cts.Token.WhenCanceled();
-                    Logger.LogInformation("Stopping timers.");
-                    timers.Stop();
-                    Logger.LogInformation("Closing connection to Edge Hub.");
-                    await moduleClient.CloseAsync();
-
-                    completed.Set();
-                    handler.ForEach(h => GC.KeepAlive(h));
-                    Logger.LogInformation("Load Gen complete. Exiting.");
-                }
+                Guid batchId = Guid.NewGuid();
+                Logger.LogInformation($"Batch Id={batchId}");
+
+                // setup the message PeriodicTask
+                messageTask = Option.Some(new PeriodicTask(
+                    () => GenerateMessageAsync(moduleClient, batchId),
+                    Settings.Current.MessageFrequency,
+                    TimeSpan.FromSeconds(Settings.Current.StartDelay),
+                    Logger,
+                    "Generate message"));
+
+                // setup the twin update PeriodicTask
+                twinUpdateTask = Option.Some(new PeriodicTask(
```

Comment:
This twin update is never checked, so do we need it?

Reply:
Same with this

Reply:
Please remove it.

Reply:
Removed all twin update logic